src/main/java/seedu/address/model/person/Analytics.java

```diff
@@ -0,0 +1,227 @@
+     * @param loan The loan to update the fields with.
+     */
+    private void updateDateFields(Loan loan) {
+        if (this.earliestLoanDate == null || loan.getStartDate().before(this.earliestLoanDate)) {
+            this.earliestLoanDate = loan.getStartDate();
+        }
+        if (this.latestLoanDate == null || loan.getStartDate().after(this.latestLoanDate)) {
+            this.latestLoanDate = loan.getStartDate();
+        }
+        if (!loan.isReturned()) {
+            if (this.earliestReturnDate == null || loan.getReturnDate().before(this.earliestReturnDate)) {
+                this.earliestReturnDate = loan.getReturnDate();
+            }
+            if (this.latestReturnDate == null || loan.getReturnDate().after(this.latestReturnDate)) {
+                this.latestReturnDate = loan.getReturnDate();
+            }
+        }
+    }
+
+    /**
+     * Returns an Analytics object that represents the analytics of a LoanRecords object.
+     * @param loanRecords The LoanRecords object to get the analytics from.
+     * @return The Analytics object that represents the analytics of the LoanRecords object.
+     */
+    public static Analytics getAnalytics(LoanRecords loanRecords) {
+        Analytics analytics = new Analytics();
+        for (int i = 0; i < loanRecords.size(); i++) {
+            Loan loan = loanRecords.getLoan(i);
+            analytics.updateNumFields(loan);
+            analytics.updateValueFields(loan);
+            analytics.updateDateFields(loan);
+        }
+        analytics.updatePropFields();
+        analytics.updateAverageFields();
+        return analytics;
+    }
+
+    public int getNumLoans() {
+        return numLoans;
+    }
+
+    public int getNumOverdueLoans() {
+        return numOverdueLoans;
+    }
+
+    public int getNumActiveLoans() {
+        return numActiveLoans;
+    }
+
+    public float getPropOverdueLoans() {
+        return propOverdueLoans;
+    }
+
+    public float getPropActiveLoans() {
+        return propActiveLoans;
+    }
+
+    public float getTotalValueLoaned() {
+        return totalValueLoaned;
+    }
+
+    public float getTotalValueOverdue() {
+        return totalValueOverdue;
+    }
+
+    public float getTotalValueActive() {
+        return totalValueActive;
+    }
+
+    public float getAverageLoanValue() {
+        return averageLoanValue;
+    }
+
+    public float getAverageOverdueValue() {
+        return averageOverdueValue;
+    }
+
+    public float getAverageActiveValue() {
+        return averageActiveValue;
+    }
+
+    public Date getEarliestLoanDate() {
+        return earliestLoanDate;
+    }
+
+    public Date getEarliestReturnDate() {
+        return earliestReturnDate;
+    }
+
+    public Date getLatestLoanDate() {
+        return latestLoanDate;
+    }
+
+    public Date getLatestReturnDate() {
+        return latestReturnDate;
+    }
+
+    @Override
+    public String toString() {
+        return "Number of loans: " + numLoans + "\n"
```

Comment:
Maybe `StringBuilder` will be more appropriate since this is a significant number of strings?

Reply:
This is fine, it's standard to just add strings like this since this is a constant operation.